components/MashDashboard/MashDashboard.spec.tsx

```diff
@@ -0,0 +1,74 @@
+import { render, screen, fireEvent } from '@testing-library/react';
+import { mockedMashReferrals } from 'factories/mashReferral';
+import MashDashboard from './MashDashboard';
+
+let tabType: string | undefined;
+let confirmationType: string | undefined;
+const mockedRouter = {
+  query: { confirmation: confirmationType, tab: tabType },
+  push: jest.fn(),
+};
+
+jest.mock('next/router', () => ({
+  useRouter: () => mockedRouter,
+}));
+
+describe('MashDashboard', () => {
+  beforeEach(() => {
+    mockedRouter.query.tab = undefined;
+    mockedRouter.query.confirmation = undefined;
+  });
+
+  it('renders the dashboard default tab to be the contact tab', () => {
+    render(<MashDashboard referrals={mockedMashReferrals} />);
+
+    expect(screen.getByText(`Contact (0)`).parentElement?.className).toContain(
+      'active'
+    );
+  });
+
+  it('selects the correct tab if the url query is updated', () => {
+    mockedRouter.query.tab = 'initial-decision';
+    render(<MashDashboard referrals={mockedMashReferrals} />);
+
+    expect(
+      screen.getByText(`Initial decision (${mockedMashReferrals.length})`)
```

Comment:
We get the number from the mocked object here but hard code zero for the contacts. Does mockedMashReferrals contain only initial decision objects. I don't think it's very clear why we call `.length` here and then hard code `0` above.

Is it possible to create the number of contacts, initial decisions, etc in the test setup and then check their numbers match in the expectation?

Reply:
Added some setup for getting the correct number of referral for each tab in `f111abe` & `c8e8736`